src/sentry/integrations/api/endpoints/data_forwarding_details.py

```diff
@@ -0,0 +1,344 @@
+        """
+        data = dict(request.data)
+        data["organization_id"] = organization.id
+
+        serializer = DataForwarderSerializer(
+            data_forwarder, data=data, context={"organization": organization}
+        )
+        if serializer.is_valid():
+            data_forwarder = serializer.save()
+            return Response(
+                serialize(data_forwarder, request.user),
+                status=status.HTTP_200_OK,
+            )
+        return self.respond(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def _validate_project_ids_input(
+        self, request: Request
+    ) -> tuple[list[int] | None, Response | None]:
+        """
+        Returns:
+            tuple: (project_ids, None) if valid,
+                   (None, error_response) if invalid with 400 Bad Request
+        """
+        raw_project_ids: Any = request.data.get("project_ids")
+        if raw_project_ids is None or not isinstance(raw_project_ids, list):
+            return None, self.respond(
+                {"project_ids": ["This field is required and must be a list"]},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        return raw_project_ids, None
+
+    def _validate_projects_exist(
+        self, projects: list[Project], project_ids: list[int]
+    ) -> Response | None:
+        """
+        Returns:
+            None if all project IDs are valid,
+            Response with 400 Bad Request if any project IDs are invalid
+        """
+        if len(projects) != len(project_ids):
+            found_ids = {project.id for project in projects}
+            invalid_ids = set(project_ids) - found_ids
+            return self.respond(
+                {
+                    "project_ids": [
+                        f"Invalid project IDs for this organization: {', '.join(map(str, invalid_ids))}"
+                    ]
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        return None
+
+    def _get_accessible_projects_to_unenroll(
+        self, request: Request, organization: Organization, enrolled_project_ids: list[int]
+    ) -> tuple[list[int] | None, Response | None]:
+        """
+        Filter enrolled projects to only include those the user has project:write access to.
+
+        Returns:
+            tuple: (accessible_project_ids, None) - list of project IDs the user can access
+                   Note: Never returns an error response, only filters the accessible projects
+        """
+        projects_to_unenroll: list[int] = []
+        for project_id in enrolled_project_ids:
+            try:
+                project = Project.objects.get(id=project_id, organization_id=organization.id)
+                if request.access.has_project_scope(project, "project:write"):
+                    projects_to_unenroll.append(project_id)
+            except Project.DoesNotExist:
+                pass
```

Comment:
Hmm, this is kinda your decision but if the person tries to unenroll projects which they dont have access to, should we let them know? Should the operation continue despite the invalid input? In other parts of this PR we do exit early (_validate_project_permissions) 🤔